contrib/kube-prometheus/assets/prometheus/rules/kubernetes.rules.yaml

```diff
@@ -84,3 +84,24 @@ groups:
     annotations:
       description: No API servers are reachable or all have disappeared from service
         discovery
+
+  - alert: K8sCertificateExpirationNotice
+    labels:
+      severity: warning
+    annotations:
+      description: Kubernetes API Certificate is expiring soon (less than 7 days)
+    expr: sum(apiserver_client_certificate_expiration_seconds_bucket{le="604800"}) > 0
+
+  - alert: K8sCertificateExpirationLastDayNotice
+    labels:
+      severity: warning
```

Comment:
I would also make this `severity: critical`, although not yet interrupting service, this will definitely be causing outages within 24 hours.